libs/community/langchain_community/tools/audio/huggingface_text_to_speech_inference.py

```diff
@@ -0,0 +1,100 @@
+import os
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+
+import requests
+from langchain_core.callbacks import CallbackManagerForToolRun
+from langchain_core.pydantic_v1 import SecretStr
+from langchain_core.tools import BaseTool
+
+
+class HuggingFaceSupportedAudioFormat(Enum):
+    WAV = WAVE = WAVEFORM = "wav"
+
+
+class HuggingFaceTextToSpeechModelInference(BaseTool):
+    """
+    HuggingFace Text-to-Speech Model Inference.
+
+    Requirements:
+
+        - Environment variable ``HUGGINGFACE_API_KEY`` must be set,
+          or passed to the constructor.
+    """
+
+    name: str = "openai_text_to_speech"
+    description: str = "A wrapper around OpenAI Text-to-Speech API. "
+
+    model: str
+    api_url: str
+    huggingface_api_key: SecretStr
+    format: HuggingFaceSupportedAudioFormat
```

Comment:
let's avoid using the enum here instead use `Literal` and do a run-time check, it saves user the trouble of having to import enums in their code.

format = Literal['wave']

Reply:
Fixed here: https://github.com/langchain-ai/langchain/pull/18880/commits/d49cf8272c50849153df49a8c7a364c6d511862a

Reply:
I just turned it into a string. The model I was exploring supported only wav, but I believe others support other formats, but wasn't able to find anything in the huggingface docs. So I think it would be easier to let the user specify rather than trying to keep up with supported formats.